src/rfc3986/_mixin.py

```diff
@@ -1,18 +1,26 @@
 """Module containing the implementation of the URIMixin class."""
+import typing as t
 import warnings
 
 from . import exceptions as exc
 from . import misc
 from . import normalizers
 from . import validators
+from ._typing_compat import Self as _Self
+
+
+class _AuthorityInfo(t.TypedDict):
+    """A typed dict for the authority info triple: userinfo, host, and port."""
+
+    userinfo: t.Optional[str]
+    host: t.Optional[str]
+    port: t.Optional[str]
 
 
 class URIMixin:
     """Mixin with all shared methods for URIs and IRIs."""
 
-    __hash__ = tuple.__hash__
```

Comment:
Right, this is something that might be outside the scope of this PR since it's "just" annotations, so I can revert it no problem. That said, I moved this line from here to within `IRIReference` (not just a straight delete) for a few reasons:

- `IRIReference` is the class that's actually subclassing a named tuple and redefining `__eq__`, thus benefiting from this.
- Imo, it makes more sense for the subclass of a tuple to be the place where __hash__ is redefined instead of within this mixin; technically, any non-tuple classes could subclass this mixin, so one could say it's more "correct" to not put this here. That being said, it is an internal class, 
- `URIReference` had defined `__hash__` within itself the same way, so I thought it would make the two reference classes more symmetrical.
- Finally, as far as I can tell, this doesn't actually provide this hashing ability to subclasses? If `__eq__` is redefined in a subclass (as the Reference classes do) and `__hash__` isn't as well, Python implicitly sets `__hash__` to None. Minimal repro in python3.8:
```py
from typing import NamedTuple
from collections import namedtuple

class Test(namedtuple("_Test", ("a", "b"))):  # Can replace this with NamedTuple for the same result.
    a: int
    b: str

class Mixin:
    __hash__ = tuple.__hash__

class SubTest(Test, Mixin):
    def __eq__(self, other: object, /) -> bool:
        if isinstance(other, type(self)):
            return self == other
        return NotImplemented

print(SubTest(1, "hello"))
hash(SubTest(1, "hello"))  # TypeError: unhashable type: 'SubTest'
```

I hope this makes sense. I'm more than willing to leave this alone for now and maybe discuss changing it another time.

Reply:
No that all makes sense. Thanks for explaining. I missed that you moved it to `IRIReference`.